worker_pool/worker_pool.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+)
+
+func main() {
+	var urls = []string{"https://www.google.com", "https://www.methods.co.uk", "https://www.github.com", "https://www.stackoverflow.com", "https://go.dev", "https://www.youtube.com", "https://www.ons.gov.uk", "https://coronavirusresources.phe.gov.uk/", "https://campaignresources.phe.gov.uk/resources", "https://www.twitter.com", "https://www.facebook.com"}
+	//because the urls will be sent to the tasks channel synchronously, the channel must be sufficiently buffered to have space to receive them (I think)
+	tasks := make(chan string, 8)
```

Comment:
following on from naming conventions - is it obvious from how this is used that it is a channel? (rather than an array of tasks or similar)

Reply:
ha, touché.  to be fair to channels it is mostly fairly obvious because they are either preceded or followed by <-, but not e.g. when they are ranged over.  anyway good point, will change

Reply:
yeah, looking at the examples, naming conventions for channels to seem to be pretty much as you've named them ('out', etc).